compiler/src/dotty/tools/dotc/transform/Splicing.scala

```diff
@@ -198,10 +200,57 @@ class Splicing extends MacroTransform:
       val newTree = transform(tree)
       val (refs, bindings) = refBindingMap.values.toList.unzip
       val bindingsTypes = bindings.map(_.termRef.widenTermRefExpr)
-      val methType = MethodType(bindingsTypes, newTree.tpe)
+      val types = bindingsTypes.collect {
+        case AppliedType(tycon, List(arg: TypeRef)) if tycon.derivesFrom(defn.QuotedTypeClass) => arg
+      }
+      val newTypeParams = types.map { tpe =>
+        newSymbol(
+          spliceOwner,
+          UniqueName.fresh(tpe.symbol.name.toTypeName),
+          Param,
+          TypeBounds.empty
```

Comment:
Shouldn't the type bounds of the type params match the type bounds of the original type for the encoding to  always typecheck?

Reply:
Seems that I am missing something. I will add some test cases and update these bounds.